adapters/sn2core/sn2core_test.go

```diff
@@ -623,3 +623,148 @@ func TestAdaptCompiledClass(t *testing.T) {
+	t *testing.T,
+	preConfirmed *core.PreConfirmed,
+	blockNum uint64,
+	response *starknet.PreConfirmedBlock,
+	expectedTxCount int,
+	expectedCandidateCount int,
+	expectedEventCount uint64,
+) {
+	assert.Equal(t, blockNum, preConfirmed.Block.Number)
+	assert.NotNil(t, preConfirmed.Block.EventsBloom)
+	assert.Empty(t, preConfirmed.Block.Hash)
+	assert.Empty(t, preConfirmed.Block.ParentHash)
+	assert.Empty(t, preConfirmed.Block.GlobalStateRoot)
+	assert.Empty(t, preConfirmed.Block.Signatures)
+	assert.Empty(t, preConfirmed.StateUpdate.NewRoot)
+	assert.Empty(t, preConfirmed.StateUpdate.BlockHash)
+	assert.Equal(t, response.Timestamp, preConfirmed.Block.Timestamp)
+	assert.Equal(t, expectedTxCount, len(preConfirmed.Block.Transactions))
+	assert.Equal(t, expectedCandidateCount, len(preConfirmed.CandidateTxs))
+	assert.Equal(t, uint64(expectedTxCount), preConfirmed.Block.TransactionCount)
+	assert.Equal(t, expectedEventCount, preConfirmed.Block.EventCount)
+	assert.Equal(t, response.Version, preConfirmed.Block.ProtocolVersion)
+	assert.Equal(t, response.L1GasPrice.PriceInFri, preConfirmed.Block.L1GasPriceSTRK)
+	assert.Equal(t, response.L1GasPrice.PriceInWei, preConfirmed.Block.L1GasPriceETH)
+}
+
+func assertCandidateTxs(
+	t *testing.T,
+	transactions []starknet.Transaction,
+	candidateTxs []core.Transaction,
+	offset int,
+) {
+	t.Helper()
+	if len(candidateTxs) > 0 {
+		for i, txn := range candidateTxs {
+			adaptedTx, err := sn2core.AdaptTransaction(&transactions[offset+i])
+			require.NoError(t, err)
+			assert.Equal(t, adaptedTx, txn)
+		}
+	}
+}
+
+func assertPreConfirmedBlockGasPrices(t *testing.T, response *starknet.PreConfirmedBlock, block *core.Block) {
+	t.Helper()
+	if response.L1DataGasPrice != nil && response.L1DataGasPrice.PriceInFri != nil {
+		assert.Equal(t, response.L1DataGasPrice.PriceInFri, block.L1DataGasPrice.PriceInFri)
+	}
+	if response.L1DataGasPrice != nil && response.L1DataGasPrice.PriceInWei != nil {
+		assert.Equal(t, response.L1DataGasPrice.PriceInWei, block.L1DataGasPrice.PriceInWei)
+	}
+	if response.L2GasPrice != nil && response.L2GasPrice.PriceInFri != nil {
+		assert.Equal(t, response.L2GasPrice.PriceInFri, block.L2GasPrice.PriceInFri)
+	}
+	if response.L2GasPrice != nil && response.L2GasPrice.PriceInWei != nil {
+		assert.Equal(t, response.L2GasPrice.PriceInWei, block.L2GasPrice.PriceInWei)
+	}
+}
+
+func countEventsAndTxs(receipts []*starknet.TransactionReceipt) (uint64, int) {
+	var evCount uint64
+	txCount := 0
+	for _, r := range receipts {
+		if r == nil {
+			break
+		}
+		evCount += uint64(len(r.Events))
+		txCount++
+	}
+	return evCount, txCount
+}
+
+func assertPreConfirmedBlockReceipts(
+	t *testing.T,
+	expectedReceipts []*starknet.TransactionReceipt,
+	blockReceipts []*core.TransactionReceipt,
+	expectedPreConfirmedTxCount int,
+) {
+	t.Helper()
+	if assert.Equal(t, expectedPreConfirmedTxCount, len(blockReceipts)) {
+		for i := range expectedPreConfirmedTxCount {
+			r := expectedReceipts[i]
```

Comment:
nit: I think this could help readability a bit.

```suggestion
		for i, actualReceipt := range blockReceipts {
			expectedReceipt := expectedReceipts[i]
```
or 
```suggestion
		for i, actual := range blockReceipts {
			expected := expectedReceipts[i]
```

Reply:
changed as suggested